### DISCUSSION TITLE
Question about proper usage of useFieldArray update

### DISCUSSION THREAD

**Discussion by jung-han** [react-hook-form/react-hook-form#9835] [Q&A]
First of all, thank you so much for making a good library. Performance could be greatly improved through react hook form. Anyway, I'm still a beginner, so please forgive me for asking wierd questions.

* sandbox: https://codesandbox.io/s/test-5lcbgi?file=/src/App.tsx

In my current project, `useFieldArray` is used as an example to manage list-type items.
As an example, some of the list values are directly update through input, and some values are updated through useFieldArray `update` method.

The problem occurs when all the relevant items are entered during update. When an item is handed by spread operator, this item is uncontrolled and does not apply the updated input value. In example

1. updating the input value
2. when the button is pressed, the input is updated with the previous value.

I thought that it could be solved by using `getValues` items, by using the watched item, or by updating in the onChange handler for inputs that were not written in the example but managed uncontrolled, but I wonder if it is a recommended method.

Is it recommended to use `useWatch` and `setValue` instead of just useFieldArray in this case?

Please let me know if there is something I misunderstood. I will wait for your reply!

**Comment by galnir** [comment]
https://github.com/react-hook-form/react-hook-form/discussions/9720#discussioncomment-4597575

**Reply by jung-han** [reply]
No. It cannot be said to be a solution, but we are developing with a consistent direction. 
Since we are using array data, we are using `useFieldArray` and making full use of the methods it provides. 
When we need to use uncontrolled input values, we are using `getValues`. I don't think it's a good way, because I have to write a code while constantly recognizing when to use getValues and when to use the provided field. But I think it's enough.

Are you experiencing similar problems?

**Reply by galnir** [reply]
I did but I just found https://github.com/react-hook-form/react-hook-form/discussions/9720#discussioncomment-4597575 to be helpful

**Reply by jung-han** [reply]
Thanks for reply! But It is the same method as I mentioned above.
It could be a solution, but I wondered if there was a method recommended by the library or recommended in terms of performance.
